src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+    ITEM_MASK_STONE,         // PLAYER_MASK_STONE
+    ITEM_MASK_BREMEN,        // PLAYER_MASK_BREMEN
+    ITEM_MASK_BLAST,         // PLAYER_MASK_BLAST
+    ITEM_MASK_SCENTS,        // PLAYER_MASK_SCENTS
+    ITEM_MASK_GIANT,         // PLAYER_MASK_GIANT
+    ITEM_MASK_FIERCE_DEITY,  // PLAYER_MASK_FIERCE_DEITY
+    ITEM_MASK_GORON,         // PLAYER_MASK_GORON
+    ITEM_MASK_ZORA,          // PLAYER_MASK_ZORA
+    ITEM_MASK_DEKU,          // PLAYER_MASK_DEKU
+};
+#endif
+
+u8 func_80122ED8(s32 maskIdMinusOne) {
+    return D_801BFDE8[maskIdMinusOne];
+}
+
+u8 Player_CurMaskToItemId(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (player->currentMask != PLAYER_MASK_NONE) {
+        return func_80122ED8(player->currentMask - 1);
+    }
+
+    return ITEM_NONE;
+}
+
+void func_80122F28(Player* player) {
+    if ((player->actor.category == ACTORCAT_PLAYER) &&
+        (!(player->stateFlags1 & (PLAYER_STATE1_20000000 | PLAYER_STATE1_800000 | PLAYER_STATE1_200000 |
+                                  PLAYER_STATE1_800 | PLAYER_STATE1_400))) &&
+        (!(player->stateFlags2 & PLAYER_STATE2_1))) {
+        if (player->doorType < 0) {
+            ActorCutscene_SetIntentToPlay(0x7C);
+        } else {
+            ActorCutscene_SetIntentToPlay(0x7D);
+        }
+    }
+}
+
+s32 func_80122F9C(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags2 & PLAYER_STATE2_80000) && player->unk_AE7 == 2;
+}
+
+s32 func_80122FCC(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return (player->stateFlags2 & PLAYER_STATE2_80000) && (player->unk_AE7 == 1 || player->unk_AE7 == 3);
+}
+
+void func_8012300C(GlobalContext* globalCtx, s32 arg1) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    player->unk_B2B = arg1;
+}
+
+void func_8012301C(Player* player, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    s32 pad;
+
+    player->unk_AE7++;
+
+    if (player->unk_AE7 == 2) {
+        s16 objectId = gLinkFormObjectIndexes[((void)0, gSaveContext.playerForm)];
+
+        gActorOverlayTable->initInfo->objectId = objectId;
+        func_8012F73C(&globalCtx->objectCtx, player->actor.objBankIndex, objectId);
+        player->actor.objBankIndex = Object_GetIndex(&globalCtx->objectCtx, GAMEPLAY_KEEP);
+    } else if (player->unk_AE7 >= 3) {
+        s32 objBankIndex = Object_GetIndex(&globalCtx->objectCtx, gActorOverlayTable->initInfo->objectId);
+
+        if (Object_IsLoaded(&globalCtx->objectCtx, objBankIndex)) {
+            player->actor.objBankIndex = objBankIndex;
+            player->actor.shape.rot.z = gSaveContext.playerForm + 1;
+            player->actor.init = func_80160AF8;
+            player->actor.update = func_80160B80;
+            player->actor.draw = func_80160BC0;
+            gSaveContext.equippedMask = 0;
+        }
+    }
+}
+
+extern s16 D_801BFE14[][18];
+
+// OoT's Player_SetBootData?
```

Comment:
I think we can just name this, seems pretty clear that it's the same function. Only problem is OoT's name isn't really right for OoT, let alone for MM. `Player_SetMovementData`, perhaps. Need a way to specify that it's for walking/running.

Reply:
I haven't named it mainly because the whole "boot data" / "boot enum" stuff doesn't really convince me, specially in MM, where there are no explicit boots like in OoT